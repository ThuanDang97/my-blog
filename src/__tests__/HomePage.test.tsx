// Components
import Home, { getStaticProps } from '@pages/index'

// Utils
import { render } from '@utils/testUtils'

// Mocks data
import { MOCK_DATA_HOME_PAGE } from '@constants/__mock__/mockData'

jest.mock('@services/restClient', () => ({
  getDataContent: jest.fn(() => {
    return MOCK_DATA_HOME_PAGE
  }),
}))

describe('Home Page', () => {
  const props = [MOCK_DATA_HOME_PAGE]

  test('Home Page matches DOM Snapshot', () => {
    const { container } = render(<Home recipes={props} />)

    expect(container).toMatchSnapshot()
  })

  test('getStaticProps should return props ', async () => {
    const result = await getStaticProps({})

    expect(result).toStrictEqual({
      props: {
        recipes: MOCK_DATA_HOME_PAGE,
      },
    })
  })
})
