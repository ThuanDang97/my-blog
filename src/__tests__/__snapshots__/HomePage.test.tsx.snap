// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page Home Page matches DOM Snapshot 1`] = `
<div>
  <header
    class="MuiBox-root css-1p2cbqg"
  >
    <div
      class="MuiBox-root css-kp6j5k"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-9tjvhq-MuiContainer-root"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1 css-1pidjnp-MuiTypography-root"
        >
          <a
            href="/"
          >
            BEAUTYONCODE
          </a>
        </h1>
        <div
          class="MuiBox-root css-7w1jqr"
        >
          <ul
            class="MuiList-root MuiList-padding css-1mq3yj5-MuiList-root"
          >
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1rk1jkp-MuiListItem-root"
            >
              <a
                href="/"
              >
                HOME
              </a>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1rk1jkp-MuiListItem-root"
            >
              <a
                href="/category/most-views"
              >
                Most views
              </a>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1rk1jkp-MuiListItem-root"
            >
              <a
                href="/blog"
              >
                Blog
              </a>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1rk1jkp-MuiListItem-root"
            >
              <a
                href="/about"
              >
                About
              </a>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1rk1jkp-MuiListItem-root"
            >
              <a
                href="/category/short-posts"
              >
                Short Posts
              </a>
            </li>
          </ul>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-15on3wz-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1unfims-MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="SearchIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div>
          loading.....
        </div>
      </div>
    </div>
  </header>
  <main
    class="MuiBox-root css-0"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
    >
      <main
        class="MuiBox-root css-0"
      >
        <section
          class="MuiBox-root css-xi606m"
        >
          <h3
            class="MuiTypography-root MuiTypography-h3 css-dp2y11-MuiTypography-root"
          >
            Code your life in an elegant way
          </h3>
          <h5
            class="MuiTypography-root MuiTypography-h5 css-12rppva-MuiTypography-root"
          >
            by Thuan Dang
          </h5>
          <img
            alt="avatar"
            data-nimg="1"
            decoding="async"
            height="200"
            loading="eager"
            src="/_next/image?url=https%3Aurl&w=640&q=75"
            srcset="/_next/image?url=https%3Aurl&w=256&q=75 1x, /_next/image?url=https%3Aurl&w=640&q=75 2x"
            style="color: transparent; border-radius: 103% 103% 103% 103%; margin-bottom: 70px;"
            width="200"
          />
        </section>
        <section
          class="MuiBox-root css-1wmpist"
        >
          <div
            class="MuiContainer-root MuiContainer-maxWidthMd css-xdmu94-MuiContainer-root"
          >
            <div
              class="MuiBox-root css-nk32ej"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-ibn7vd-MuiTypography-root"
              >
                Xin ch√†o!
              </p>
              <div
                class="MuiBox-root css-19yfvn4"
              />
            </div>
          </div>
        </section>
      </main>
    </div>
  </main>
</div>
`;
